### PR TITLE
Fix ConvTranspose behavior when stride, padding > 1

### PR DESCRIPTION
**Pull request by Hyunseok-Kim0**
### 1. Content and background
https://numbersmithy.com/understanding-transposed-convolutions-in-pytorch/#org9cb8239
After some searching, I found that extra slice layer is required to implement same action of pytorch, onnx `ConvTranspose` in tflite.

https://github.com/PINTO0309/onnx2tf/blob/6b971b8d836cf263763458ff2c18152c9bcfe6e5/onnx2tf/ops/ConvTranspose.py#L180-L187

Also, I found the reason of segmentation fault during invoke. Segmention fault is occured when wrong output shape is assigned to `conv_func`. I added extra code to calculate exact shape of `ConvTranspose` of tensorflow to prevent this.

### 2. Summary of corrections
Now the tensorflow output shape is calculated before setting `pad_mode`. `pad_mode` is always set to `VALID` and extra `slice` layer is added after `ConvTranspose` layer if needed.

ONNX | Tflite
--- | --- 
| 

I used code below to debug various combination of kernel, stride, padding.
~~However, in some cases like kernel = 3, stride = 1, padding = 1, `ConvTranspose` operation is converted to `FlexConv2DBackpropInput` operation. I tried but couldn't find the reason yet. It seems to occur when calculated padding in one direction does not match to other side when pad_mode is `SAME`. (ex. left padding=3, right padding=2)~~
Flex operation is removed by fixing pad_mode to always `VALID`.

```
import os
import shutil

import onnxruntime
import tensorflow as tf
import torch
from einops import rearrange
from torch import nn
import numpy as np
from onnx2tf import convert


class dummy_network(nn.Module):

    def __init__(self, kernel, stride, padding, groups):
        super(dummy_network, self).__init__()

        self.tconv = torch.nn.ConvTranspose2d(16, 64, kernel, stride=stride, padding=padding, groups=groups)

    def forward(self, x):
        result = self.tconv(x)

        return result


def main():
    os.makedirs("tflite", exist_ok=True)
    np.random.seed(12)
    kernels = [np.random.randint(3, 8) for _ in range(30)]
    strides = [np.random.randint(1, 8) for _ in kernels]
    paddings = [np.random.randint(1, 8) for _ in kernels]
    groups = np.random.choice([1, 2, 4, 8], size=len(kernels))

    for k, s, p, g in zip(kernels, strides, paddings, groups):
        print("kernel, stride, padding, groups : ", k, s, p, g)
        model = dummy_network(k, s, p, g)
        model.eval()

        dummy_name = "dummy_tconv"
        onnx_save_path = f"tflite/{dummy_name}.onnx"
        temp_tflite = "tflite/model_float32.tflite"
        tflite_save_path = f"tflite/{dummy_name}.tflite"

        # y1, x1, y2, x2
        dummy_x = torch.randn(1, 16, 224, 224, dtype=torch.float32)

        torch.onnx.export(model,
                          args=(dummy_x,),
                          f=onnx_save_path,
                          input_names=["x"],
                          output_names=["result"],
                          opset_version=11)

        # get onnx output
        # -----------------------------------------------------------------------------------------------
        onnx_session = onnxruntime.InferenceSession(onnx_save_path, providers=['CPUExecutionProvider'])
        onnx_inputs = dict(x=dummy_x.numpy())
        onnx_outputs = onnx_session.run(None, onnx_inputs)

        convert(onnx_save_path, output_folder_path="tflite")
        shutil.move(temp_tflite, tflite_save_path)

        # get tflite output
        # -----------------------------------------------------------------------------------------------
        tflite_onnx2tf = tf.lite.Interpreter(model_path=tflite_save_path)
        tflite_onnx2tf.allocate_tensors()
        tflite_onnx2tf.set_tensor(tflite_onnx2tf.get_input_details()[0]['index'],
                                  rearrange(dummy_x.numpy(), "n c h w -> n h w c"))
        tflite_onnx2tf.invoke()
        tflite_output = [tflite_onnx2tf.get_tensor(i['index']) for i in tflite_onnx2tf.get_output_details()][0]

        np.testing.assert_allclose(rearrange(tflite_output, "n h w c -> n c h w"), onnx_outputs[0], atol=1e-3, rtol=0.5)
        print("onnx and tflite have same output")

    print("convert done")

    return


if __name__ == "__main__":
    main()


```

### 4. Issue number (only if there is a related issue)
#56

### CODE REVIEW

**Comment by PINTO0309** [comment]
Thanks for always issuing useful pull requests.

Assign a `Ready to merge` label when ready to merge. It would be very helpful as a guide as to when I press the merge button. :)

image

**Comment by Hyunseok-Kim0** [comment]
It looks I don't have permission to add a label to PR. This PR is ready to be merged after current check is completed without problem.

**Comment by PINTO0309** [comment]
I learned that only the repository owner and collaborators can be assigned labels.
